### DISCUSSION TITLE
Generating Skull Surface from CHARM Output

### DISCUSSION THREAD

**Discussion by DSliva** [simnibs/simnibs#427] [7.Creating head models]
Hello, 

I'd like to use the skull segmentation from SimNIBS to create BEM (FreeSurfer) surfaces for the forward model in MNE-Python. So far, I've had some success using the following: 

1. Run CHARM to get skull volume segmentation (in final_tissues.nii.gz or tissue_labeling_upsampled.nii.gz)
2. Isolate skull segmentation using mri_binarize (FreeSurfer)
3. Convert binarized volume to SimNIBS mesh using meshmesh (SimNIBS)
4. Convert SimNIBS mesh to FreeSurfer surface using simnibs.write_freesurfer_surface (SimNIBS)

However, this gives only outer skull (not inner skull, which I need), and the surface has several holes in it. Does anyone know if there is an easier/more efficient way to generate a skull surface from CHARM output? 

Thank you, 
Danielle

**Comment by tmedani** [comment]
Hi @DSliva 

From our experience [within [Brainstorm ](https://github.com/brainstorm-tools/brainstorm3)], the BEM computation is not possible with a realistic skull shape. 
BEM solvers assume smoother BEM surfaces. 
This is also the recommendation of the BEM developers [OpenMEEG & MNE]

If you want to use the realistic FEM model from SImNibs, you can either refer to this [paper](10.1016/j.neuroimage.2023.120259)


or use the SimNibs mesh within [Brainstorm FEM forward modeling](https://neuroimage.usc.edu/brainstorm/Tutorials/FemMedianNerveCharm) 

A+
Takfarinas

**Comment by jduemose** [comment]
Hi Danielle

If you want to use the simnibs output, I too, like Takfarinas, would suggest you to do the leadfield calculations using FEM rather than BEM. We do provide a way to convert a TDCS simulation from simnibs to a forward model object in MNE-Python. Currently, it works with MNE <= 1.5 as far as I remember (there are some caveats, e.g., when it comes to dipole fitting). I am currently writing a small tutorial on how to do this but the paper referenced above also includes the basics. If you decide to proceed with simnibs and have problems, let us know, otherwise the Brainstorm community is very helpful as well.

Best,
Jesper
